src/context/WalletProvider/WalletContext.tsx

```diff
@@ -7,6 +7,7 @@ import type { DeviceState, InitialState, KeyManagerWithProvider } from './Wallet
 
 export interface IWalletContext {
   state: InitialState
+  getAdapter: (keyManager: KeyManager, index?: number) => Promise<any>
```

Comment:
can we avoid any type here?

Reply:
Not for now, see https://github.com/shapeshift/web/pull/5466#discussion_r1359948979

Reply:
Discussed this one out, will leave it for a part 4 PR in the stack since this will require an hdwallet bumpy